backend/templates/create-api/index.js

```diff
@@ -1,65 +1,119 @@
-                console.log(consoleSuccess, `Success! ${templateTypes[i]} created!`),
-                  console.log(consoleClear, "");
-                process.exit(1);
-              }
-              fs.appendFile(
-                `src/${templateTypes[i]}/${databaseTableName}.js`,
-                variableReplacements,
-                "utf8",
-                (err) => {
-                  if (err) return console.error(consoleError, err), console.log(consoleClear, "");
-                }
-              );
-            });
-          });
-        } else {
-          //If API files do exist, notify user in the command line and exit
-          console.log("test");
-          console.error(
-            consoleError,
-            `Files for ${databaseTableName} API already exist, please try again with a new name`
-          ),
-            console.log(consoleClear, ""),
-            createAPIFiles();
-        }
-      });
-    }
-  );
+/**
+ * Generic console error message
+ * @param {*} message
+ */
+const sayError = (message) => {
+  console.error(`${consoleErrorColor}${message}${consoleClearColors}`);
+};
+
+/**
+ * Generic console success message
+ * @param {*} message
+ */
+const saySuccess = (message) => {
+  console.log(`${consoleSuccessColor}${message}${consoleClearColors}`);
 };
 
-createAPIFiles();
+/**
+ * This function asks the user what the name of their database table is. Once the users inputs a value in the console and presses enter the script either creates the appropriate API files or
+ * errors because the files already exists.  If there is an error because the files already exist, the script will ask again to enter the name of a database table to build the API for.
+ */
+class createAPIFiles {
+  constructor() {
+    this.databaseTableName = "";
+    this.askTheQuestion();
+  }
+
+  askTheQuestion() {
+    readline.question(
+      "\n What is the name of the database table you'd like to create an API for? ",
+      this.handleQuestionResponse.bind(this)
+    );
+  }
+
+  handleQuestionResponse(databaseTableName) {
+    this.databaseTableName = databaseTableName;
+
+    if (this.checkFilesAlreadyExist()) {
+      sayError(
+        `Files for "${databaseTableName}" API endpoint already exist, please try again with a new name.`
+      );
+      this.askTheQuestion();
+    } else {
+      // We are done asking questions, close the readline so that we don't wait for more input.
+      readline.close();
+      this.generateNewFiles();
+    }
+  }
+
+  checkFilesAlreadyExist() {
+    for (const templateType of templateTypes) {
+      if (fs.existsSync(`src/${templateType}/${this.databaseTableName}.js`)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  generateNewFiles() {
+    const mapObj = { $databaseTableName: this.databaseTableName };
+
+    // Oldschool function declaration syntax, because we need to bind(), below.
+    templateFilePaths.forEach(
+      function (templateFilePath, i) {
```

Comment:
Should be an arrow function.

Reply:
how do you .bind() on an arrow function?